### DISCUSSION TITLE
Reason for dark mode toggle positioning?

### DISCUSSION THREAD

**Discussion by leroyvn** [jpanther/congo#393] [General]
First of all, thanks for this excellent theme. I transitioned my project website to it a few months ago and working with it has so far been a real pleasure.

Now, the question: What is the motivation for positioning the dark mode toggle in the footer? 

My actual point is: on short pages, where no scrolling is required, it stands out quite well; however, on websites with longer pages, finding it is more challenging.

**Comment by rshmhrj** [comment]
Interesting question @leroyvn!
I personally have a set-it-and-forget-it mindset regarding dark mode.  If a user comes in, doesn't like what they see, they can switch and then all other pages should be in the same mode.  That functionality is probably only used once -- I assume that's why the choice was to put it in the footer rather than the header.

I looked at a few other [themes](https://hugothemes.netlify.app/tags/dark-mode/) and there doesn't seem to be any standard or consensus regarding where it should be placed.  On new sites, I think I naturally just scan all four sides to see if there is a toggle (and if I can't find it, then would go into settings / preferences and see if it is there).

@jpanther I looked at @leroyvn's site and noticed a weird spacing issue on semi-long content though -- the scroll-to-top arrow displays below the footer: https://www.eradiate.eu/site/post/20211018-fall_news/


![image](https://user-images.githubusercontent.com/750874/204614502-68817943-c783-4e76-8e23-33ae68a47bad.png)

On short pages, it doesn't display.  On long pages, they line up perfectly.  On this specific page, there's a bunch of space underneath the footer and the scroll-to-top ends up looking weird.  Should we open an issue for this?

**Comment by jpanther** [comment]
The reasoning for not having the toggle more visible is because it isn't really a primary feature. The reality is that you shouldn't need a toggle like this on your site at all. Most websites bury this type of preference in a settings page or similar because the operating system should (and does) manage this preference on behalf of the user.

How many websites do you go to and attempt to toggle light/dark appearance? I just don't think it's an important enough feature to have it taking up valuable real estate higher up the page. I feel like the visual clutter distracts from the minimal and clean nature of the theme.

**Reply by leroyvn** [reply]
Thanks @jpanther for the explanation.

**Comment by jpanther** [comment]
Further update on this... I'm pleased to report that v2.5.0 of the theme now includes the option to position the appearance switcher within any menu.

**Reply by orlandoflv** [reply]
Hi @jpanther, is there a way to toggle the icons between "sun" and "moon"?  I mean here:

```
[[main]]
  identifier = "appearance"
  weight = 50
  [main.params]
    action = "appearance"
    icon = "sun"
```

**Reply by orlandoflv** [reply]
Never mind, I found that I just needed to remove the last line. Thank you very much for this awesome theme!
